pkg/i2gw/providers/common/utils.go

```diff
@@ -17,24 +17,104 @@ limitations under the License.
 package common
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"log"
 	"regexp"
+	"strings"
 
 	networkingv1 "k8s.io/api/networking/v1"
 	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
 )
 
-func GetIngressClass(ingress networkingv1.Ingress) string {
-	var ingressClass string
+var k8sClientset *kubernetes.Clientset
+
+func init() {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(errors.New("unable to create config"))
+	}
+
+	k8sClientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
```

Comment:
This init function will always try to create a k8s client, even if the user doesn't have a cluster and want to convert some local file.

You'll need to pass the user's choice of where he wants to read resources from (cluster or file) and the k8s client, which was initialized in the i2gw package.